src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonValueConverterOfT.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    // Used for value converters that need to re-enter the serializer since it will
+    // support JsonPath and other features that require per-serialization state.
+    internal abstract class JsonValueConverter<T> : JsonConverter<T>
```

Comment:
I was deferring discussing naming until we discuss the public side. However, I do think `JsonValueConverter` is descriptive and all future converters can just use this one instead of directly using `JsonConverter`.

It may also not return a `bool` for the read\write -- so these classes need to be refactored once we approve the API. Also, I didn't bother trying to rename `JsonReadState` etc which is a parameter to Read\Write to something more public-friendly (e.g. `JsonDeserializationState` for this PR.

Also there will likely be at least two other public converters -- one for collections and one for objects -- which will be new and not in this PR since they will require different interaction\callbacks that what we use internally.